src/marqo/tensor_search/tensor_search.py

```diff
@@ -1929,45 +1934,104 @@ def _vector_text_search(
+                                    }
+                                },
+                                "query": {
+                                    "function_score": {
+                                        "query": {
+                                            "knn": {
+                                                f"{TensorField.chunks}.{vector_field}": {
+                                                    "vector": vectorised_text,
+                                                    "k": result_count + offset
+                                                }
+                                            }
+                                        },
+                                        "functions": [
+                                            {
+                                                "script_score": {
+                                                    "script": {
+                                                        "source": script_score
+                                                    }
+                                                }
+                                            }
+                                        ],
+                                        "boost_mode": "replace"
+                                    }
+                                },
+                                "score_mode": "max"
                             }
-                        }
-                    },
-                    "score_mode": "max"
+                        },
+                    }
+                },
+                "_source": {
+                    "exclude": ["__chunks.__vector_*"]
                 }
-            },
-            "_source": {
-                "exclude": ["__chunks.__vector_*"]
             }
-        }
 
-        field_names = list(index_info.get_text_properties().keys())
-        if attributes_to_retrieve is not None:
-            search_query["_source"] = {"include": attributes_to_retrieve} if len(attributes_to_retrieve) > 0 else False
+            if attributes_to_retrieve is not None:
+                search_query["_source"] = {"include": attributes_to_retrieve} if len(attributes_to_retrieve) > 0 else False
 
-        if filter_string is not None:
-            search_query["query"]["nested"]["query"]["knn"][f"{TensorField.chunks}.{vector_field}"][
-                "filter"] = {
-                "query_string": {"query": f"{contextualised_filter}"}
+            if filter_string is not None:
+                search_query["query"]["function_score"]["query"]["nested"]["query"]["function_score"]["query"]\
+                    ["knn"][f"{TensorField.chunks}.{vector_field}"][
+                    "filter"] = {
+                    "query_string": {"query": f"{contextualised_filter}"}
+                }
+            body += [{"index": index_name}, search_query]
+    else:
+        for vector_field in vector_properties_to_search:
```

Comment:
Same as above - move this into a new function

Reply:
done